ArenaEnemyCooldown.lua

```diff
@@ -0,0 +1,290 @@
+
+    -- Find the icon to use
+    if self.icons[spellId] then
+        NS.StartWeakAuraIcon(self.icons[spellId]);
+    end
+end
+
+local function ProcessUnitSpellCast(self, event, ...)
+    local guid = ValidateUnit(self);
+    if ( not guid ) then return end
+
+    local unitTarget, _, spellID = ...;
+    if ( unitTarget == self.unit ) then
+        local spell = spellData[spellID];
+        if ( not spell ) or ( spell.trackEvent ~= "UNIT_SPELLCAST_SUCCEEDED" ) then return end
+        if self.icons[spellID] then
+            NS.StartWeakAuraIcon(self.icons[spellID]);
+        end
+    end
+end
+
+local function ProcessUnitAura(self, event, ...)
+    local guid = ValidateUnit(self);
+    if ( not guid ) then return end
+
+    local unitTarget, updateAuras = ...;
+    -- Only use UNIT_AURA to extend aura
+    if ( unitTarget == self.unit ) and updateAuras and updateAuras.updatedAuraInstanceIDs then
+        for _, instanceID in ipairs(updateAuras.updatedAuraInstanceIDs) do
+            local spellInfo = C_UnitAuras.GetAuraDataByAuraInstanceID(unitTarget, instanceID);
+            if spellInfo then
+                local spellID = spellInfo.spellId
+                local spell = spellData[spellID]
+                if ( not spell ) or ( not spell.extend ) or ( not self.activeMap[spellID] ) then return end
+                NS.RefreshWeakAuraDuration(self.activeMap[spellID]);
+            end
+        end
+    end
+end
+
+local function ArenaEventHandler(self, event, ...)
+    if ( event == "COMBAT_LOG_EVENT_UNFILTERED" ) then
+        ProcessCombatLogEvent(self, event, ...);
+    elseif ( event == "UNIT_SPELLCAST_SUCCEEDED" ) then
+        ProcessUnitSpellCast(self, event, ...);
+    elseif ( event == "UNIT_AURA" ) then
+        ProcessUnitAura(self, event, ...);
+    end
+end
+
+-- Premake all icons (regardless of class)
+local premadeIcons = {};
+if test then
+    local unitId = "player";
+    premadeIcons[unitId] = {};
+    for spellID, spell in pairs(spellData) do
+        premadeIcons[unitId][spellID] = NS.CreateWeakAuraIcon(unitId, spellID, 32, true);
+    end
+else
+    for i = 1, NS.MAX_ARENA_SIZE do
+        local unitId = "arena"..i;
+        premadeIcons[unitId] = {};
+        for spellID, spell in pairs(spellData) do
+            premadeIcons[unitId][spellID] = NS.CreateWeakAuraIcon(unitId, spellID, 32, true);
+        end
+    end
+end
+
+local function GetUnitSpec(unit)
+    if ( unit == "player" ) then
+        local currentSpec = GetSpecialization();
+        return GetSpecializationInfo(currentSpec);
+    else
+        local arenaIndex = string.sub(unit, -1, -1);
+        return GetArenaOpponentSpec(index);
+    end
+end
+
+local function SetupAuraGroup(group, unit)
+    -- Clear previous icons
+    NS.IconGroup_Wipe(group);
+
+    if ( not UnitExists(unit) ) then return end
+
+    local class = select(3, UnitClass(unit));
+    -- Pre-populate icons
+    for spellID, spell in pairs(spellData) do
```

Comment:
Buttons are not populated in arena skirmishes

Reply:
Cannot check UnitExists, since buttons are populated in preparation phase, when UnitExists will return false.

Should check class ~= nil

Reply:
UnitClass returns nil when unit is not in range as well.
Should just check spec and translate to class.

Reply:
Latest iteration, icons are populated but not showing.

Reply:
GetSpecializationInfoByID returns class name, not index.

Reply:
Latest iteration uses classFileName